lib/curl_multibyte.c

```diff
@@ -82,6 +82,21 @@ char *curlx_convert_wchar_to_UTF8(const wchar_t *str_w)
 
 #if defined(USE_WIN32_LARGE_FILES) || defined(USE_WIN32_SMALL_FILES)
 
+int curlx_win32_open(const char *filename, int oflag)
+{
+#ifdef _UNICODE
+  int result = 0;
```

Comment:
The error value is defined as -1 and 0 might be a valid result: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen?view=msvc-160#return-value

Reply:
Thanks, totally missed it (and the later check as mentioned below).